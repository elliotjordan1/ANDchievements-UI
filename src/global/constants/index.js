export const ANDi = 'ANDi';
export const Client = 'Client';
export const TechStack = 'Tech Stack';