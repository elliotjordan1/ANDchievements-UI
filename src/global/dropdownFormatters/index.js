/* eslint-disable import/prefer-default-export */
import { getClients } from './clients';

export {
  getClients
}