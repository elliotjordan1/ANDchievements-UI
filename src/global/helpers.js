export const onInputChange = set => event => set(event.target.value);

export const urlValidator = (givenValue) => {
  return (/^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([-.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/.test((givenValue)));
}

export const shuffleAndSliceArray = (array, sliceLength) => {
  let newArray = array;

  while (newArray === array) {
    const arrayLength = array.length;

    const shuffledArray = array.sort(() => Math.random() - 0.5);
    
    newArray = shuffledArray.slice(0, arrayLength > sliceLength ? sliceLength : arrayLength);
  }

  return newArray;
}

export default onInputChange;
