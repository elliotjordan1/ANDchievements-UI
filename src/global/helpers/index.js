import { onInputChange } from './inputs';
import { shuffleAndSliceArray } from './arrays';

export {
  onInputChange,
  shuffleAndSliceArray
}