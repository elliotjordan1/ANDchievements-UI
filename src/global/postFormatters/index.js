/* eslint-disable import/prefer-default-export */
import ProjectPostFormatter from './project';

export {
  ProjectPostFormatter
}