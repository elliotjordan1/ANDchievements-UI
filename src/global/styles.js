import styled from 'styled-components';
import theme from './theme';

export const HomepageWrapper = styled.div`
  align-content: row;
  background-color: ${theme.colours.brand.base};
  min-height: calc(100vh - 78px);
  @media ${theme.media.tablet}{
    display: flex;
    flex-wrap: wrap;
    min-height: calc(100vh - 114px);
  }
  @media ${theme.media.desktop}{
    min-height: calc(100vh - 153px);

  }
`;

export const TextWrapper = styled.div`
  align-self: flex-end;
  position: relative;
  width: 100%;
`;

export const ErrorWrapper = styled(HomepageWrapper)`
  color: ${theme.colours.brand.red};
  font-family: ${theme.fontFamily.primary};
  font-size: 22px;
  font-weight: 300;
  justify-content: center;
  line-height: 20;
  text-align: center;
`;

export const FormWrapper = styled.div`
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: white;
  @media ${theme.media.tablet}{
    margin: 25px 30px;
    height: min-content;
  }
  @media ${theme.media.desktop}{
    margin: 45px 70px;
  }
`;