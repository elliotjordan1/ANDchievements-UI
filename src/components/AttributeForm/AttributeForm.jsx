/* eslint-disable import/extensions */
import React, { useState, useReducer } from 'react';
import PropTypes from 'prop-types';
import { useToasts } from 'react-toast-notifications';
import formReducer, { initialFormState } from '../../reducers/createProject/reducer';
import { onInputChange } from '../../global/helpers';
import { SubmitButton, Label, FormInput, FormWrapper } from './AttributeFormStyles.js';
import ANDiCreator from '../../api/handlers/attributeCreation/andi';
import TechStackCreator from '../../api/handlers/attributeCreation/techStack';
import ClientCreator from '../../api/handlers/attributeCreation/client';

const AttributeForm = ({
  formType
}) => {
  const { addToast, removeAllToasts } = useToasts();  

  const [imageUrl, setImageUrl] = useState('');
  const [name, setName] = useState('');

  const [state, dispatch] = useReducer(formReducer, initialFormState);

  const creationStrategies = [
    new ANDiCreator(),
    new TechStackCreator(),
    new ClientCreator()
  ];

  const handleSubmit = async (event) => {
    event.preventDefault();

    creationStrategies.forEach(async (strategy) => {
      if (strategy.appliesTo(formType)) {
        const response = await strategy.create({
          name,
          imageUrl
        });     

        if (response.status === 201) {
          const { data } = response;

          addToast(`Success! ${name} has been added, with ImageURL: ${imageUrl}`, { appearance: 'success' });   

          dispatch({ type: `ADD_${formType}`, data });       
          console.log(state);   
        } else {
          addToast('Uh oh! Something went wrong!', { appearance: 'error' });
        }
      }      
    });

    setTimeout(() => {
      removeAllToasts();
    }, 3000);    
  };

  const nameLabel = `${formType  } Name`;
  const imageUrlLabel = `${formType  } Image URL`;

  return (
    <>
      <FormWrapper onSubmit = {handleSubmit}>
        <div>
          <Label>{nameLabel}</Label>
          <FormInput maxLength = {40} placeholder = {nameLabel} value = {name} onChange={onInputChange(setName)} />
        </div>
        <div>
          <Label>{imageUrlLabel}</Label>
          <FormInput maxLength = {40} placeholder = {imageUrlLabel} value = {imageUrl} onChange={onInputChange(setImageUrl)} />
        </div>
        <div>
          <SubmitButton type = 'submit'>SUBMIT</SubmitButton>
        </div>
      </FormWrapper>
    </>
  )
}

AttributeForm.propTypes = {
  formType: PropTypes.string.isRequired
}

export default AttributeForm;