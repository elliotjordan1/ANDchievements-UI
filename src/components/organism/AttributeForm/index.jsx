/* eslint-disable import/extensions */
import React, { useState, useReducer } from 'react';
import PropTypes from 'prop-types';
import { useToasts } from 'react-toast-notifications';
import formReducer, { initialFormState } from '../../../reducers/createProject/reducer';
import { onInputChange } from '../../../global/helpers';
import { FormLabel, SubmitButton, FormInput } from '../../atom';
import { createAndi as ANDiCreator, createTechStack as TechStackCreator, createClient as ClientCreator } from '../../../api/handlers/attributeCreation';

const AttributeForm = ({
  formType
}) => {
  const { addToast } = useToasts();  

  const [imageUrl, setImageUrl] = useState('');
  const [name, setName] = useState('');

  const [, dispatch] = useReducer(formReducer, initialFormState);

  const creationStrategies = [
    new ANDiCreator(),
    new TechStackCreator(),
    new ClientCreator()
  ];

  const handleSubmit = async (event) => {
    event.preventDefault();

    creationStrategies.forEach(async (strategy) => {
      if (strategy.appliesTo(formType)) {
        const response = await strategy.create({
          name,
          imageUrl
        });     

        if (response.status === 201) {
          const { data } = response;

          addToast(`Success! ${name} has been added, with ImageURL: ${imageUrl}`, { appearance: 'success' });   

          dispatch({ type: `ADD_${formType}`, data });       
        } else {
          addToast('Uh oh! Something went wrong!', { appearance: 'error' });
        }
      }      
    });
  };

  const nameLabel = `${formType  } Name`;
  const imageUrlLabel = `${formType  } Image URL`;

  return (
    <div>
      <div>
        <FormLabel labelText={nameLabel} />
        <FormInput maxLength = {40} placeholder = {nameLabel} value = {name} onChange={onInputChange(setName)} />
      </div>
      <div>
        <FormLabel labelText = {imageUrlLabel} />
        <FormInput maxLength = {40} placeholder = {imageUrlLabel} value = {imageUrl} onChange={onInputChange(setImageUrl)} />
      </div>
      <div>
        <SubmitButton type = 'submit' text = {`Create New ${formType}`} onClick={handleSubmit} />
      </div>
    </div>
  )
}

AttributeForm.propTypes = {
  formType: PropTypes.string.isRequired
}

export default AttributeForm;