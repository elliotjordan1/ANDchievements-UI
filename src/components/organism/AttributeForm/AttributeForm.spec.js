import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ToastProvider } from 'react-toast-notifications';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import AttributeForm from '.';
import * as AttributeTypes from '../../../global/constants';

describe('AttributeForm component', () => {
  it('renders correctly for a client', () => {
    const wrapper = render(<ToastProvider><AttributeForm formType={AttributeTypes.Client} /></ToastProvider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly for a Tech Stack', () => {
    const wrapper = render(<ToastProvider><AttributeForm formType={AttributeTypes.TechStack}/></ToastProvider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly for an ANDi', () => {
    const wrapper = render(<ToastProvider><AttributeForm formType={AttributeTypes.ANDi}/></ToastProvider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('executes the correct strategy for creation of a client', () => {
    const Mock = new MockAdapter(axios);
  
    const responseObject = {
      new_client: [
        { 
          clientid: 1,
        }
      ]
    }

    Mock.onPost().reply(201, responseObject);

    const { getByText, getByPlaceholderText} = render(<ToastProvider><AttributeForm formType={AttributeTypes.Client}/></ToastProvider>);

    const nameInput = getByPlaceholderText(`${AttributeTypes.Client} Name`);
    const imageUrlInput = getByPlaceholderText(`${AttributeTypes.Client} Image URL`);
    const input = getByText('SUBMIT');

    expect(nameInput).toBeDefined();
    expect(imageUrlInput).toBeDefined();
    expect(input).toBeDefined();

    fireEvent.change(nameInput, { target: { value: 'Bean'}});
    fireEvent.change(imageUrlInput, { target: { value: 'bean.url'}})

    fireEvent.click(input);
  });

  it('executes the correct strategy for creation of an andi', () => {
    const Mock = new MockAdapter(axios);
  
    const responseObject = {
      new_andi: [
        { 
          andiid: 1,
        }
      ]
    }

    Mock.onPost().reply(201, responseObject);

    const { getByText, getByPlaceholderText} = render(<ToastProvider><AttributeForm formType={AttributeTypes.ANDi}/></ToastProvider>);

    const nameInput = getByPlaceholderText(`${AttributeTypes.ANDi} Name`);
    const imageUrlInput = getByPlaceholderText(`${AttributeTypes.ANDi} Image URL`);
    const input = getByText('SUBMIT');

    expect(nameInput).toBeDefined();
    expect(imageUrlInput).toBeDefined();
    expect(input).toBeDefined();

    fireEvent.change(nameInput, { target: { value: 'Bean'}});
    fireEvent.change(imageUrlInput, { target: { value: 'bean.url'}})

    fireEvent.click(input);
  });

  it('executes the correct strategy for creation of an andi', () => {
    const Mock = new MockAdapter(axios);
  
    const responseObject = {
      new_tech: [
        { 
          technologyid: 1,
        }
      ]
    }

    Mock.onPost().reply(201, responseObject);

    const { getByText, getByPlaceholderText} = render(<ToastProvider><AttributeForm formType={AttributeTypes.TechStack}/></ToastProvider>);

    const nameInput = getByPlaceholderText(`${AttributeTypes.TechStack} Name`);
    const imageUrlInput = getByPlaceholderText(`${AttributeTypes.TechStack} Image URL`);
    const input = getByText('SUBMIT');

    expect(nameInput).toBeDefined();
    expect(imageUrlInput).toBeDefined();
    expect(input).toBeDefined();

    fireEvent.change(nameInput, { target: { value: 'Bean'}});
    fireEvent.change(imageUrlInput, { target: { value: 'bean.url'}})

    fireEvent.click(input);
  });

  it('errors when creation of attribute fails', () => {
    const Mock = new MockAdapter(axios);
  
    Mock.onPost().networkError();

    const { getByText, getByPlaceholderText} = render(<ToastProvider><AttributeForm formType={AttributeTypes.TechStack}/></ToastProvider>);

    const nameInput = getByPlaceholderText(`${AttributeTypes.TechStack} Name`);
    const imageUrlInput = getByPlaceholderText(`${AttributeTypes.TechStack} Image URL`);
    const input = getByText('SUBMIT');

    expect(nameInput).toBeDefined();
    expect(imageUrlInput).toBeDefined();
    expect(input).toBeDefined();

    fireEvent.change(nameInput, { target: { value: 'Bean'}});
    fireEvent.change(imageUrlInput, { target: { value: 'bean.url'}})

    fireEvent.click(input);
  });
});
