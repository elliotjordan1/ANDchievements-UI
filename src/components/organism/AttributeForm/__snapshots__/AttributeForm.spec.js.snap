// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeForm component renders correctly for a Tech Stack 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="FormContainer-sc-8s9ya3-0 fDwXTj"
      >
        <div
          class="FormTitle-oi1tko-0 AttributeFormTitle-sc-1ev8f59-0 guzjNX"
        >
          Add New 
          Tech Stack
        </div>
        <div>
          <div
            class="FormLabel-wqrwj6-0 bOWOhD"
          />
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="Tech Stack Name"
            type=""
            value=""
          />
        </div>
        <div>
          <div
            class="FormLabel-wqrwj6-0 bOWOhD"
          />
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="Tech Stack Image URL"
            type=""
            value=""
          />
        </div>
        <div>
          <button
            class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
            type="submit"
          >
            Create New Tech Stack
          </button>
        </div>
      </div>
    </div>
    <div
      class="react-toast-notifications__container css-mrh2d-ToastContainer"
    />
  </body>,
  "container": <div>
    <div
      class="FormContainer-sc-8s9ya3-0 fDwXTj"
    >
      <div
        class="FormTitle-oi1tko-0 AttributeFormTitle-sc-1ev8f59-0 guzjNX"
      >
        Add New 
        Tech Stack
      </div>
      <div>
        <div
          class="FormLabel-wqrwj6-0 bOWOhD"
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="Tech Stack Name"
          type=""
          value=""
        />
      </div>
      <div>
        <div
          class="FormLabel-wqrwj6-0 bOWOhD"
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="Tech Stack Image URL"
          type=""
          value=""
        />
      </div>
      <div>
        <button
          class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
          type="submit"
        >
          Create New Tech Stack
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AttributeForm component renders correctly for a client 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="FormContainer-sc-8s9ya3-0 fDwXTj"
      >
        <div
          class="FormTitle-oi1tko-0 AttributeFormTitle-sc-1ev8f59-0 guzjNX"
        >
          Add New 
          Client
        </div>
        <div>
          <div
            class="FormLabel-wqrwj6-0 bOWOhD"
          />
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="Client Name"
            type=""
            value=""
          />
        </div>
        <div>
          <div
            class="FormLabel-wqrwj6-0 bOWOhD"
          />
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="Client Image URL"
            type=""
            value=""
          />
        </div>
        <div>
          <button
            class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
            type="submit"
          >
            Create New Client
          </button>
        </div>
      </div>
    </div>
    <div
      class="react-toast-notifications__container css-mrh2d-ToastContainer"
    />
  </body>,
  "container": <div>
    <div
      class="FormContainer-sc-8s9ya3-0 fDwXTj"
    >
      <div
        class="FormTitle-oi1tko-0 AttributeFormTitle-sc-1ev8f59-0 guzjNX"
      >
        Add New 
        Client
      </div>
      <div>
        <div
          class="FormLabel-wqrwj6-0 bOWOhD"
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="Client Name"
          type=""
          value=""
        />
      </div>
      <div>
        <div
          class="FormLabel-wqrwj6-0 bOWOhD"
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="Client Image URL"
          type=""
          value=""
        />
      </div>
      <div>
        <button
          class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
          type="submit"
        >
          Create New Client
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AttributeForm component renders correctly for an ANDi 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="FormContainer-sc-8s9ya3-0 fDwXTj"
      >
        <div
          class="FormTitle-oi1tko-0 AttributeFormTitle-sc-1ev8f59-0 guzjNX"
        >
          Add New 
          ANDi
        </div>
        <div>
          <div
            class="FormLabel-wqrwj6-0 bOWOhD"
          />
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="ANDi Name"
            type=""
            value=""
          />
        </div>
        <div>
          <div
            class="FormLabel-wqrwj6-0 bOWOhD"
          />
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="ANDi Image URL"
            type=""
            value=""
          />
        </div>
        <div>
          <button
            class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
            type="submit"
          >
            Create New ANDi
          </button>
        </div>
      </div>
    </div>
    <div
      class="react-toast-notifications__container css-mrh2d-ToastContainer"
    />
  </body>,
  "container": <div>
    <div
      class="FormContainer-sc-8s9ya3-0 fDwXTj"
    >
      <div
        class="FormTitle-oi1tko-0 AttributeFormTitle-sc-1ev8f59-0 guzjNX"
      >
        Add New 
        ANDi
      </div>
      <div>
        <div
          class="FormLabel-wqrwj6-0 bOWOhD"
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="ANDi Name"
          type=""
          value=""
        />
      </div>
      <div>
        <div
          class="FormLabel-wqrwj6-0 bOWOhD"
        />
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="ANDi Image URL"
          type=""
          value=""
        />
      </div>
      <div>
        <button
          class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
          type="submit"
        >
          Create New ANDi
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
